File: benchmarks/CWE-327-WeakCrypto/weak-crypto.js
// Vulnerable: Weak cryptographic algorithms
const crypto = require('crypto');

function hashPassword(password) {
  // Vulnerable: MD5 is broken for password hashing
  return crypto.createHash('md5').update(password).digest('hex');
}

function generateChecksum(data) {
  // Vulnerable: SHA1 is deprecated
  return crypto.createHash('sha1').update(data).digest('hex');
}

function encryptData(data, key) {
  // Vulnerable: DES is weak
  const cipher = crypto.createCipher('des', key);
  let encrypted = cipher.update(data, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return encrypted;
}

File: benchmarks/CWE-327-WeakCrypto/weak-crypto.py
# Vulnerable: Weak cryptographic algorithms
import hashlib
from Crypto.Cipher import DES

def hash_password(password):
    # Vulnerable: MD5 for password hashing
    return hashlib.md5(password.encode()).hexdigest()

def encrypt_data(data, key):
    # Vulnerable: DES encryption
    cipher = DES.new(key[:8].encode(), DES.MODE_ECB)
    padded = data + '\x00' * (8 - len(data) % 8)
    return cipher.encrypt(padded.encode()).hex()

def generate_token(user_id):
    # Vulnerable: predictable token using MD5
    return hashlib.md5(f"token-{user_id}".encode()).hexdigest()
